// ng2-dropdown-treeview/src/treeview-item.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as _ from 'lodash';
import { TreeviewItem } from './treeview-item';
var TreeviewItemComponent = (function () {
    function TreeviewItemComponent() {
        this.checkedChange = new EventEmitter();
    }
    TreeviewItemComponent.prototype.toggleCollapseExpand = function () {
        this.item.collapsed = !this.item.collapsed;
    };
    TreeviewItemComponent.prototype.onCheckedChange = function (checked) {
        if (!_.isNil(this.item.children)) {
            this.item.children.forEach(function (child) { return child.setCheckedRecursive(checked); });
        }
        this.checkedChange.emit(checked);
    };
    TreeviewItemComponent.prototype.onChildCheckedChange = function (child, checked) {
        if (this.item.checked !== checked) {
            var itemChecked = true;
            for (var i = 0; i < this.item.children.length; i++) {
                if (!this.item.children[i].checked) {
                    itemChecked = false;
                    break;
                }
            }
            this.item.checked = itemChecked;
        }
        this.checkedChange.emit(checked);
    };
    return TreeviewItemComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", TreeviewItem)
], TreeviewItemComponent.prototype, "item", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], TreeviewItemComponent.prototype, "checkedChange", void 0);
TreeviewItemComponent = __decorate([
    Component({
        selector: 'leo-treeview-item',
        template: "\n<div class=\"treeview-item\" [class.treeview-parent]=\"item.children\">\n    <i *ngIf=\"item.children\" (click)=\"toggleCollapseExpand()\" aria-hidden=\"true\"\n        class=\"fa\" [class.fa-caret-right]=\"item.collapsed\" [class.fa-caret-down]=\"!item.collapsed\"></i>\n    <label class=\"form-check-label\">\n        <input type=\"checkbox\" class=\"form-check-input\"\n            [(ngModel)]=\"item.checked\" (ngModelChange)=\"onCheckedChange($event)\" [disabled]=\"item.disabled\" />\n        {{item.text}}\n    </label>\n    <div *ngIf=\"!item.collapsed\">\n        <leo-treeview-item *ngFor=\"let child of item.children\" [item]=\"child\" (checkedChange)=\"onChildCheckedChange(child, $event)\">\n        </leo-treeview-item>\n    </div>\n</div>\n    ",
        styles: ["\n.treeview-item {\n    padding-left: 1rem;\n    white-space: nowrap;\n}\n.treeview-item .form-check-label {\n    padding-top: 2px;\n    padding-bottom: 2px;\n}\n.treeview-item .fa {\n    margin-left: -1.0rem;\n    width: 10px;\n    cursor: pointer;\n}\n    "]
    })
], TreeviewItemComponent);
export { TreeviewItemComponent };
